feat(firebase): add Google Account auth (#52)
remix-run/examples: firebase/README.md

## Running locally

> To run this example, you either need to create a firebase proejct or use the emulators:

Reviewer: ```suggestion
To run this example, you either need to create a Firebase project or use [the emulators](https://firebase.google.com/docs/emulator-suite):
```

> To run this example, you either need to create a firebase proejct or use the emulators:

Author: ```suggestion
To run this example, you either need to create a firebase project or use the emulators:
```
```suggestion
To run this example, you either need to create a firebase proejct or use the emulators:
```

> To run this example, you either need to create a Firebase project or use the emulators:

Reviewer: ```suggestion
To run this example, you either need to create a Firebase project or use [the emulators](https://firebase.google.com/docs/emulator-suite):
```

> ### 1. Run against a Firebase Project

Reviewer: ```suggestion
### 1. Run against a Firebase project
```

> 1. [Create a Firebase Project](https://console.firebase.google.com).

Reviewer: ```suggestion
1. [Create a Firebase project](https://console.firebase.google.com).
```

> 2. Enable Auth (with email) and Firestore.

Reviewer: ```suggestion
2. Enable [Auth](https://firebase.google.com/docs/auth) (with email) and [Cloud Firestore](https://firebase.google.com/docs/firestore).
```

3. Add a Web App with hosting.
4. Copy `.env.example` to `.env`.

> 5. Set the API_KEY value to the [Web API Key](https://firebase.google.com/docs/reference/rest/auth), which can be obtained on the project settings page in your Firebase admin console.

Reviewer: ```suggestion
5. Set the `API_KEY` value to the [Web API Key](https://firebase.google.com/docs/reference/rest/auth), which can be obtained on the project settings page in your Firebase admin console.
```

> 6. Set SERVICE_ACCOUNT to the contents of your service account's private key JSON file:

Reviewer: ```suggestion
6. Set `SERVICE_ACCOUNT` to the contents of your service account's private key JSON file:
```

## Deploying

> 1. Follow the "Run against a Firebase Project" steps above if not done already

Reviewer: ```suggestion
1. Follow the "Run against a Firebase project" steps above if not done already
```

2. Log in to the CLI with `npm run firebase -- login`
3. Run `npm run firebase -- use --add` and choose the Firebase project you want to deploy to
4. Deploy with `npm run firebase -- deploy`

## Integration with Google Sign-in Provider

> The "Login with Google" link will only work when running against a Firebase Project.

Reviewer: ```suggestion
The "Login with Google" link will only work when running against a Firebase project.
```

> After the steps in "Run against a Firebase Project" have been completed:

Reviewer: ```suggestion
After the steps in "Run against a Firebase project" have been completed:
```

> - In the [Firebase Console](https://console.firebase.google.com), navigate to Authentication > Sign-in method > Add new provider > Google. Make a note of the client ID and secret and add them to the .env file as GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET.

Reviewer: ```suggestion
- In the [Firebase Console](https://console.firebase.google.com), navigate to Authentication > Sign-in method > Add new provider > Google. Make a note of the client ID and secret and add them to the `.env` file as `GOOGLE_CLIENT_ID` and `GOOGLE_CLIENT_SECRET`.
```
